contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,952 @@
+             * given our liquidity.
+             *
+             * The least amount of tokens extraditable from the position is where the active trading price is
+             * at the ticker 0 meaning the pool is offering 1:1 trades between WETH & OETHb. At that moment the pool
+             * consists completely of OETHb and no WETH.
+             *
+             * The more swaps from WETH -> OETHb happen on the pool the more the price starts to move towards the -1
+             * ticker making OETHb (priced in WETH) more expensive.
+             */
+            (uint256 wethAmount, uint256 oethbAmount) = helper
+                .getAmountsForLiquidity(
+                    sqrtRatioX96TickClosestToParity, // sqrtRatioX96
+                    sqrtRatioX96Tick0, // sqrtRatioAX96
+                    sqrtRatioX96Tick1, // sqrtRatioBX96
+                    liquidity
+                );
+
+            require(wethAmount == 0, "Non zero wethAmount");
+            underlyingAssets = oethbAmount;
+        }
+    }
+
+    /**
+     * @notice Deposit all supported assets in this strategy contract to the platform
+     */
+    function depositAll() external override onlyVault nonReentrant {
+        _deposit(WETH, IERC20(WETH).balanceOf(address(this)));
+    }
+
+    /**
+     * @notice Withdraw an `amount` of assets from the platform and
+     * send to the `_recipient`.
+     * @param _recipient         Address to which the asset should be sent
+     * @param _asset             Address of the asset
+     * @param _amount            Units of asset to withdraw
+     */
+    function withdraw(
+        address _recipient,
+        address _asset,
+        uint256 _amount
+    ) external override onlyVault nonReentrant {
+        require(_asset == WETH, "Unsupported asset");
+        require(_recipient == vaultAddress, "Only withdraw to vault allowed");
+
+        uint256 wethBalance = IERC20(WETH).balanceOf(address(this));
+        if (wethBalance < _amount) {
+            uint256 additionalWethRequired = _amount - wethBalance;
+            (uint256 wethInThePool, ) = getPositionPrincipal();
+
+            if (wethInThePool < additionalWethRequired) {
+                revert NotEnoughWethLiquidity(
+                    wethInThePool,
+                    additionalWethRequired
+                );
+            }
+
+            uint256 shareOfWethToRemove = additionalWethRequired.divPrecisely(
+                wethInThePool
+            ) + 1;
+            _removePartialLiquidity(shareOfWethToRemove);
+        }
+
+        // burn remaining OETHb
+        _burnOethbOnTheContract();
+        _withdraw(_recipient, _amount);
+    }
+
+    /**
+     * @notice Withdraw all supported assets from platform and
+     * sends to the OToken's Vault.
+     */
+    function withdrawAll() external override onlyVault nonReentrant {
+        _removeAllLiquidity();
```

Comment:
🔴 This function assumes that the strategy always has some liquidity on the pool and the NFT deposited on the Gauge. This might prevent `IVault.withdrawAllFromStrategies` when there's no liquidity provided to the pool. 

A simple `tokenId > 0` check would fix this

Reply:
thanks great point. Fixed here: https://github.com/OriginProtocol/origin-dollar/pull/2184/commits/209bd39612fdbd7fade6f9c02bf313819cd402d9